### PR TITLE
Read/Write observation sets to HDF5.

### PR DESCRIPTION
**Pull request by akleeman**
This change will allow us to do things like:
- Parse RINEX/SBP observations, compute satellite state and save to HDF5.
- Run observation sets through a filter and save the observations and position solutions to HDF5
- Iterate through data stored in an HDF5 file the same way we would with observations directly from RINEX/SBP.

Also includes:
 - More thorough testing using pytest.parameterize.
 - Observation sets now include info dictionaries.
 - Solution thread returns DataFrames instead of dicts to allow serialization to HDF5.

@swift-nav/estimation

### CODE REVIEW

File: gnss_analysis/filters/kalman_filter.py [hunk 1]
```diff
@@ -542,17 +540,17 @@ def process_model(self):
...
     F[:self.ambiguity_states_idx, :self.ambiguity_states_idx] = F_dynamics
     # process noise
-    Q = np.eye(n) * self.sig_z**2
-    Q_dynamics = self.sig_dynamics**2 \
-                 * np.array(np.bmat([[(self.dt**5 / 20) * np.eye(self.n_dim), (self.dt**4 / 8) * np.eye(self.n_dim), (self.dt**3 / 6) * np.eye(self.n_dim)],
-                                     [(self.dt**4 / 8) * np.eye(self.n_dim), (self.dt**3 / 3) * np.eye(self.n_dim), (self.dt**2 / 2) * np.eye(self.n_dim)],
-                                     [(self.dt**3 / 6) * np.eye(self.n_dim), (self.dt**2 / 2) * np.eye(self.n_dim), self.dt * np.eye(self.n_dim)]]))
+    Q = np.eye(n) * self.sig_z ** 2
+    Q_dynamics = self.sig_dynamics ** 2 \
+                 * np.array(np.bmat([[(self.dt ** 5 / 20) * np.eye(self.n_dim), (self.dt ** 4 / 8) * np.eye(self.n_dim), (self.dt ** 3 / 6) * np.eye(self.n_dim)],
```
Comment (benjamin0) [inline gnss_analysis/filters/kalman_filter.py]: The whitespace is nice, but it almost looks like `self.dt ^ (5/20)` at first glance

Comment (akleeman) [inline gnss_analysis/filters/kalman_filter.py]: Ah yeah sorry. I have a PEP8 checker that automatically changed those.  I agree it's misleading and worth breaking convention.

File: tests/conftest.py [hunk 2]
```diff
@@ -128,3 +128,55 @@ def rinex_base(datadir):
 def rinex_navigation(datadir):
   basename = 'short_baseline_cors/seat032/seat0320.16n'
   return datadir.join(basename).strpath
+
```
Comment (benjamin0) [inline tests/conftest.py]: are these ^ fixtures needed anymore?

File: tests/test_filters.py [hunk 3]
```diff
@@ -78,83 +79,79 @@ def assert_matches_at_some_point():
...
       print "python", baseline
       print "actual", locations.NOVATEL_BASELINE
 
+
 @pytest.mark.slow
 @pytest.mark.regression
-@pytest.mark.parametrize('filter_class', [kalman_filter.StaticKalmanFilter,
-                                          kalman_filter.DynamicKalmanFilter])
-def test_cors_baseline(datadir, filter_class):
+def test_cors_baseline(cors_short_baseline, dgnss_filter_class):
```
Comment (benjamin0) [inline tests/test_filters.py]: will this parametrize over dgnss_filter_class?

File: tests/test_filters.py [hunk 4]
```diff
@@ -164,35 +161,30 @@ def eventually_close():
 
 @pytest.mark.slow
 @pytest.mark.regression
-def test_cors_drops_reference(datadir, dgnss_filter):
+def test_cors_drops_reference(cors_drops_reference, dgnss_filter_class):
```
Comment (benjamin0) [inline tests/test_filters.py]: this docstring was copied from the other test. probably should update it.

File: tests/conftest.py (hunk 2 above)
Comment (akleeman) [inline tests/conftest.py]: `test_rinex.py` still uses them ... but could probably be made not to.

File: tests/test_filters.py (hunk 3 above)
Comment (akleeman) [inline tests/test_filters.py]: Yep, I defined a new fixture in `conftest.py` that contains all filters we want tested.

File: tests/test_filters.py (hunk 4 above)
Comment (akleeman) [inline tests/test_filters.py]: good catch

Comment (akleeman) [inline tests/test_filters.py]: I actually ended up just merging the tests together.

**Pull request merged**